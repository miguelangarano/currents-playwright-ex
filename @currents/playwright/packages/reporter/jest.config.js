module.exports = {
  verbose: true,
  clearMocks: true,
  testEnvironment: "node",
  testMatch: ["<rootDir>/**/__tests__/**/?(*.)(spec|test).ts"],
  transform: {
    "^.+\\.tsx?$": "@swc/jest",
  },
  moduleNameMapper: {
    "@currents/reporter/(.*)$": "<rootDir>/../reporter/$1",
  },
  transformIgnorePatterns: ["node_modules"],
  moduleFileExtensions: ["ts", "js", "d.ts"],
  coverageReporters: ["json", "lcov", "text"],
};
