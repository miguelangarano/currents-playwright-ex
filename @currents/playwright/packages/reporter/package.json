{
  "name": "@currents/playwright",
  "version": "0.11.5",
  "main": "./dist/index.js",
  "author": "Currents Software Inc",
  "license": "GPL-3.0-or-later",
  "scripts": {
    "test": "jest",
    "rm": "rimraf dist",
    "lint": "TIMING=1 eslint \"**/*.ts*\"",
    "build": "tsup-node --minify --dts",
    "dev": "tsup-node --watch --dts",
    "release": "release-it ",
    "publish:npm": "npm run rm  && npm run build && ./publish.js",
    "release-ci": "npm run rm && npm run build && ./publish.js -t latest"
  },
  "bin": {
    "pwc": "./dist/bin/index.js"
  },
  "devDependencies": {
    "@playwright/test": "1.40.0",
    "@release-it/conventional-changelog": "^7.0.2",
    "@swc/jest": "^0.2.24",
    "@types/async-retry": "^1.4.5",
    "@types/babel__code-frame": "^7.0.3",
    "@types/debug": "^4.1.11",
    "@types/getos": "^3.0.1",
    "@types/jest": "^29.5.3",
    "@types/json-stringify-safe": "^5.0.0",
    "@types/lodash": "^4.14.191",
    "@types/pluralize": "^0.0.33",
    "@types/randomstring": "^1.1.8",
    "@types/shelljs": "^0.8.11",
    "@types/stack-utils": "^2.0.1",
    "eslint": "^7.32.0",
    "eslint-config-custom": "*",
    "jest": "^29.5.0",
    "msw": "^1.2.3",
    "release-it": "^16.2.1",
    "rimraf": "^4.1.1",
    "tsconfig": "*",
    "tsup": "^6.5.0",
    "typescript": "^4.5.2",
    "wtfnode": "^0.9.1"
  },
  "dependencies": {
    "@babel/code-frame": "^7.18.6",
    "@commander-js/extra-typings": "^11.1.0",
    "@currents/commit-info": "file:./@currents/commit-info",
    "@currents/pwc-parser": "*",
    "@currents/pwc-scanner": "*",
    "async-retry": "^1.3.3",
    "axios": "^1.2.0",
    "axios-retry": "^3.4.0",
    "chalk": "^4.1.2",
    "colors": "^1.4.0",
    "commander": "^11.1.0",
    "date-fns": "^2.29.3",
    "debug": "^4.3.4",
    "dotenv": "^16.0.3",
    "execa": "^7.2.0",
    "getos": "^3.2.1",
    "image-size": "^1.0.2",
    "json-stringify-safe": "^5.0.1",
    "lodash": "^4.17.21",
    "nanoid": "^3.3.4",
    "p-all": "^3.0.0",
    "pino": "^8.11.0",
    "pluralize": "^8.0.0",
    "pretty-ms": "^7.0.1",
    "randomstring": "^1.2.3",
    "source-map-support": "^0.5.21",
    "stack-utils": "^2.0.6",
    "tmp-promise": "^3.0.3",
    "ts-pattern": "^4.3.0"
  },
  "files": [
    "dist",
    "!**/*.map",
    "!**/*/*.map",
    "**/*.LEGAL.txt",
    "LICENSE.md",
    "README.md",
    "CHANGELOG.md"
  ],
  "keywords": [
    "playwright",
    "playwright dashboard",
    "cloud dashboard",
    "reporter",
    "currents",
    "tests reporter",
    "sorry cypress",
    "ci"
  ],
  "exports": {
    ".": {
      "import": "./dist/index.js",
      "require": "./dist/index.js"
    },
    "./parallel": {
      "import": "./dist/parallel/index.js",
      "require": "./dist/parallel/index.js"
    },
    "./package.json": "./package.json"
  },
  "publishConfig": {
    "access": "public",
    "registry": "https://registry.npmjs.org"
  },
  "release-it": {
    "npm": {
      "publish": false
    },
    "github": {
      "release": true
    },
    "plugins": {
      "@release-it/conventional-changelog": {
        "preset": "angular",
        "header": "# Changelog",
        "ignoreVersion": true,
        "infile": "../../CHANGELOG.md"
      }
    },
    "git": {
      "commitMessage": "chore: release v${version}",
      "tagName": "v${version}"
    }
  }
}
