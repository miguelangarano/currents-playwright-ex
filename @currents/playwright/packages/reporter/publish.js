#!/usr/bin/env node

const { execSync } = require("child_process");
const fs = require("fs");
const pkg = require("./package.json");
const { Command } = require("commander");

const program = new Command()
  .name("publish")
  .option("-t, --tag <beta | latest>", "npm dist-tag to publish to");

program.parse(process.argv);
const options = program.opts();

console.log(options);
if (!options.tag) {
  console.log("No tag supplied: beta or latest");
  process.exit(1);
}
console.log(process.cwd());
fs.copyFileSync("../../CHANGELOG.md", "./CHANGELOG.md");
fs.copyFileSync("../../LICENSE.md", "./LICENSE.md");
execSync(`npm pack --dry-run && npm publish --tag ${options.tag}`, {
  cwd: "./",
  stdio: "inherit",
});
