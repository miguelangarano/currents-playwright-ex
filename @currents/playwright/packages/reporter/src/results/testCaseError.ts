import { codeFrameColumns } from "@babel/code-frame";

import {
  TestError as PlaywrightTestError,
  TestCase,
  TestResult,
  TestStep,
} from "@playwright/test/reporter";
import { FullProject } from "@playwright/test/types/test";
import fs from "fs";
import { last, orderBy } from "lodash";
import path from "path";
import { stripAnsiEscapes } from "../lib/stripAnsiEscapes";
import { TestError } from "./types";

const interruptedError: TestError = {
  message: "Test interrupted",
  name: "Error",
  stack: "",
  codeFrame: undefined,
};

// get display error for a test case with miltiple attempts
export function getTestCaseDisplayError(testCase: TestCase) {
  if (!testCase.results?.length) {
    return interruptedError.message;
  }

  const orderedAttemptResults = orderBy(testCase.results, "retry", "asc");

  // if there is at least one failed attempt, use the last one
  const result = last(
    orderedAttemptResults.filter(
      (r) =>
        r.status === "failed" ||
        r.status === "timedOut" ||
        r.status === "interrupted"
    )
  )?.error?.message;

  return result ? stripAnsiEscapes(result) : null;
}

export function getTestAttemptError(result: TestResult, project?: FullProject) {
  if (result.status === "interrupted") {
    return interruptedError;
  }
  if (!result.error) {
    return null;
  }

  if (result.error) {
    return getErrorDetails(result.error, project);
  }

  const errorStep = tryGetStepWithErrorAndLocation(result.steps);
  if (!errorStep) return null;
  if (!errorStep?.error) {
    errorStep.error = result.error;
  }

  return {
    message: getErrorMessage(errorStep.error),
    name: errorStep.title,
    stack: getErrorStack(errorStep.error),
    codeFrame: testStepToCodeFrame(errorStep),
  };
}

export function getErrorDetails(
  error: PlaywrightTestError,
  project?: FullProject
) {
  try {
    return {
      message: getErrorMessage(error),
      name: getErrorName(error),
      stack: getErrorStack(error),
      codeFrame:
        error.location &&
        getErrorCodeFrame({
          location: error.location,
          snippet: error.snippet,
          project,
        }),
    };
  } catch (e) {
    return {
      message: "Error while getting error details",
      name: "Error",
      stack: "",
      codeFrame: undefined,
    };
  }
}
function getErrorName(error: PlaywrightTestError) {
  return "Error";
}

function getErrorCodeFrame({
  location,
  snippet,
  project,
}: {
  location: NonNullable<PlaywrightTestError["location"]>;
  snippet: PlaywrightTestError["snippet"];
  project?: FullProject;
}): TestError["codeFrame"] {
  return {
    line: location.line,
    column: location.column,
    originalFile: location.file,
    relativeFile: project?.testDir
      ? path.relative(project.testDir, location.file)
      : null,
    absoluteFile: location.file,
    frame: stripAnsiEscapes(snippet ?? "No error source code detected"),
    language: getLanguageFromExtension(location.file),
  };
}

function tryGetStepWithErrorAndLocation(
  steps: TestStep[],
  depth = 3
): TestStep | undefined {
  let result;

  for (const s of steps) {
    if (s.error && s.location) {
      result = s;
    }

    if (result) return result;

    if (s.steps.length > 0 && depth > 0) {
      result = tryGetStepWithErrorAndLocation(s.steps, depth - 1);
    }
  }

  for (const s of steps) {
    if (s.category === "hook" && s.location) {
      result = s;
    }

    if (result) return result;

    if (s.steps.length > 0 && depth > 0) {
      result = tryGetStepWithErrorAndLocation(s.steps, depth - 1);
    }
  }

  return result;
}

function getErrorMessage(error?: PlaywrightTestError) {
  return typeof error === "string"
    ? error
    : stripAnsiEscapes(error?.message ?? "");
}

function getErrorStack(error?: PlaywrightTestError) {
  return error !== undefined && "stack" in error && error.stack
    ? error.stack
    : "";
}

function testStepToCodeFrame(
  step: TestStep
): TestError["codeFrame"] | undefined {
  const { location } = step;
  if (!location) {
    return undefined;
  }

  const language = getLanguageFromExtension(location.file);
  const absoluteFile = location.file;
  const relativeFile = path.relative(process.cwd(), location.file);
  const originalFile = relativeFile;

  const column = location.column;
  const line = location.line;
  const source = getSourceFromLocation(location.file);
  const frame = source
    ? codeFrameColumns(source, { start: { line, column } })
    : "";

  return {
    line,
    column,
    originalFile,
    relativeFile,
    absoluteFile,
    frame,
    language,
  };
}

const getLanguageFromExtension = (filePath: string) => {
  return (path.extname(filePath) || "").toLowerCase().replace(".", "");
};

const getSourceFromLocation = (filePath: string) => {
  try {
    const source = fs.readFileSync(filePath, "utf-8") + "\n//";
    return source;
  } catch (e) {
    return undefined;
  }
};
