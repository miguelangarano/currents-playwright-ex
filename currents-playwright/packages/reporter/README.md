# @currents/playwright

Playwright integration and reporting tool for [Currents](https://currents.dev/playwright) - a cloud dashboard for debugging, troubleshooting and analysing parallel CI tests supporting Cypress and Playwright.

<p align="center">
  <img width="830" src="https://static.currents.dev/currents-playwright-banner-gh.png" />
</p>

<p align="center">
<a href="https://currents.dev/readme">Documentation</a> | <a href="https://currents.dev?utm_source=currents-playwright-readme">Currents</a>

</p>

- Saves traces, videos, screenshots and console output to a cloud
- Fetches git information and associated with CI builds
- Integrates with your workflow - Slack, GitHub or GitLabPR comments and status checks
- Flakiness, failure rate, duration and much more aggregative metrics
- Powerful history and trends browser on test or spec level
- Common errors tracker
- Automated reports with test suite health metrics
- REST API and HTTP Webhooks

---

## Install

Make sure you already have Playwright installed, then run:

```sh
npm install @currents/playwright
```

## Enable traces, screenshots and videos

```js
use: {
  // ...
  trace: "on",
  video: "on",
  screenshot: "on",
}
```

## Usage

Choose the preferred launch method:

- executing a `pwc` CLI command - it runs playwright with a predefined configuration
- add `@currents/playwright` reporter to Playwright configuration file

### `pwc` CLI

We need to pass three parameters to run `pwc`:

- our [record key](https://currents.dev/readme/guides/record-key)
- the project ID, which is created when you create a project in the Current dashboard
- the [CI build ID](https://currents.dev/readme/guides/ci-build-id)

The command passes down all the other CLI flags to the Playwright test runner as-is. We can pass these as command line arguments, as environment variables, or a mixture of both.

```sh
pwc --project-id PROJECT_ID --key RECORD_KEY --ci-build-id hello-currents --tag tagA,tagB
```

### `@currents/playwright` reporter

Alternatively, you can manually add the reporter to playwright configuration and keep using `playwright test` CLI command.

```ts
import type { PlaywrightTestConfig } from "@playwright/test";
import { currentsReporter } from '@currents/playwright';

const currentsConfig = {
  ciBuildId: process.env.CURRENTS_CI_BUILD_ID,
  recordKey: process.env.CURRENTS_RECORD_KEY,
  projectId: process.env.CURRENTS_PROJECT_ID,
  tag: ["runTagA", "runTagB"],
};

const config: PlaywrightTestConfig = {
  // explicitly provide reporter name and configuration
  [
    "@currents/playwright",
    currentsConfig
  ],
  // ...or use the helper function that ensures type safety
  currentsReporter(currentsConfig),
  /* other reporters, if exist, e.g.:
  ["html"]
  */
};

export default config;
```

You can also provide configuration by setting environment variables before running `playwright` command

```sh
CURRENTS_RECORD_KEY=RECORD_KEY CURRENTS_PROJECT_ID=PROJECT_ID CURRENTS_CI_BUILD_ID=hello-currents CURRENTS_TAG=tagA,tagB npx playwright test
```

## Examples

Check out the example repository for integrating with GitHub Actions: https://github.com/currents-dev/playwright-gh-actions-demo, including parallel runs on multiple machines using [Playwright shards](https://playwright.dev/docs/test-parallel#shard-tests-between-multiple-machines).

Hereâ€™s a quick reference to configuration files:

- [test-basic-pwc.yml](https://github.com/currents-dev/playwright-gh-actions-demo/blob/main/.github/workflows/test-basic-pwc.yml) - using `pwc` executable script
- [test-basic-reporter.yml](https://github.com/currents-dev/playwright-gh-actions-demo/blob/main/.github/workflows/test-basic-reporter.yml) - using reporter configuration

## Screenshots

By default Playwright only captures screenshots at the end of a test, according to the provided [screenshot](https://playwright.dev/docs/screenshots) option. Manually created screenshots are hidden by default and not attached to any test.

To send additional screenshots to Currents, they have to be attached to the test. For example, you can attach a screenshot to a test like this

```js
const { test, expect } = require("@playwright/test");

test("basic test", async ({ page }, testInfo) => {
  await page.goto("https://playwright.dev");
  const screenshot = await page.screenshot();
  await testInfo.attach("screenshot", {
    body: screenshot,
    contentType: "image/png",
  });
});
```

For more information see the Playwright [test info attachment](https://playwright.dev/docs/api/class-testinfo#test-info-attach) documentation.

## Requirements

- NodeJS 14.0.0+
- Playwright 1.22.2+
