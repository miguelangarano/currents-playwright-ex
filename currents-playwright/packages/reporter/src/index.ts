import "source-map-support/register";

import { CurrentsConfig } from "./config/config";
import { DefaultReporter } from "./reporters";

export type { CurrentsConfig } from "./config/config";
/**
 * Create {@link https://currents.dev/playwright | Currents Reporter} to be used with playwright test runner
 *
 * @augments CurrentsConfig
 * @param {CurrentsConfig} config - Currents Reporter {@link https://currents.dev/readme/integration-with-playwright/currents-playwright | configuration}
 * @returns {["@currents/playwright", CurrentsConfig]} a tuple of reporter name and config
 */
export const currentsReporter = (config: CurrentsConfig) =>
  ["@currents/playwright" as string, config] as [string, CurrentsConfig];

export default DefaultReporter;
