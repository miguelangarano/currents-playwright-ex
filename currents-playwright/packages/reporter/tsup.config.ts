import { defineConfig } from "tsup";

export default defineConfig({
  entry: ["src/index.ts", "src/bin/index.ts"],
  esbuildOptions: (options) => {
    options.legalComments = "linked";
  },
  splitting: false,
  shims: true,
  clean: true,
  sourcemap: true,
  platform: "node",
  target: "esnext",
});
