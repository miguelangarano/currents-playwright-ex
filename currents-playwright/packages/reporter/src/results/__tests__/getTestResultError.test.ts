import { describe, expect, it } from "@jest/globals";
import fs from "fs";
import path from "path";

import { TestCase } from "@playwright/test/reporter";
import { getTestAttemptError } from "../testCaseError";

const fixtures = [
  "beforeAll error.json",
  "beforeAll timeout.json",
  "beforeEach error.json",
  "beforeEach timeout.json",
  "expect timeout.json",
  "global timeout A.json",
  "global timeout B.json",
  "global timeout explicitly skipped test.json",
];

describe("Getting error details", () => {
  it.each(fixtures)("should fetch error from '%s'", (key) => {
    const testCase = JSON.parse(
      fs.readFileSync(path.join(__dirname, "fixtures", key), "utf8")
    ) as unknown as TestCase;

    expect(getTestAttemptError(testCase.results[0])).toMatchSnapshot();
  });
});
