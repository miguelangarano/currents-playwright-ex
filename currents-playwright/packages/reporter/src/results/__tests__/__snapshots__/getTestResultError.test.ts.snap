// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Getting error details should fetch error from 'beforeAll error.json' 1`] = `
{
  "codeFrame": {
    "absoluteFile": "/Users/agoldis/currents-playwright/apps/e2e/errorDetails/beforeAll.error.spec.ts",
    "column": 11,
    "frame": "  4 | test.describe(title, function () {
  5 |   test.beforeAll(async () => {
> 6 |     throw new Error("beforeAll error");
    |           ^
  7 |   });
  8 |   test(title, async function ({ page }) {
  9 |     await page.goto("/");",
    "language": "ts",
    "line": 6,
    "originalFile": "/Users/agoldis/currents-playwright/apps/e2e/errorDetails/beforeAll.error.spec.ts",
    "relativeFile": null,
  },
  "message": "beforeAll error",
  "name": "Error",
  "stack": "Error: beforeAll error
    at /Users/agoldis/currents-playwright/apps/e2e/errorDetails/beforeAll.error.spec.ts:6:11",
}
`;

exports[`Getting error details should fetch error from 'beforeAll timeout.json' 1`] = `
{
  "codeFrame": {
    "absoluteFile": "/Users/agoldis/currents-playwright/apps/e2e/errorDetails/beforeAll.timeout.spec.ts",
    "column": 8,
    "frame": "  4 | const t = "beforeAll timeout";
  5 | test.describe(t, function () {
> 6 |   test.beforeAll(async () => {
    |        ^
  7 |     await wait(11000);
  8 |   });
  9 |   test(t, async function ({ page }) {",
    "language": "ts",
    "line": 6,
    "originalFile": "/Users/agoldis/currents-playwright/apps/e2e/errorDetails/beforeAll.timeout.spec.ts",
    "relativeFile": null,
  },
  "message": ""beforeAll" hook timeout of 1000ms exceeded.",
  "name": "Error",
  "stack": "[31m"beforeAll" hook timeout of 1000ms exceeded.[39m
    at /Users/agoldis/currents-playwright/apps/e2e/errorDetails/beforeAll.timeout.spec.ts:6:8",
}
`;

exports[`Getting error details should fetch error from 'beforeEach error.json' 1`] = `
{
  "codeFrame": {
    "absoluteFile": "/Users/agoldis/currents-playwright/apps/e2e/errorDetails/beforeEach.error.spec.ts",
    "column": 11,
    "frame": "  3 | test.describe(t, function () {
  4 |   test.beforeEach(async () => {
> 5 |     throw new Error("beforeEach error");
    |           ^
  6 |   });
  7 |   test(t, async function ({ page }) {
  8 |     await page.goto("/");",
    "language": "ts",
    "line": 5,
    "originalFile": "/Users/agoldis/currents-playwright/apps/e2e/errorDetails/beforeEach.error.spec.ts",
    "relativeFile": null,
  },
  "message": "beforeEach error",
  "name": "Error",
  "stack": "Error: beforeEach error
    at /Users/agoldis/currents-playwright/apps/e2e/errorDetails/beforeEach.error.spec.ts:5:11",
}
`;

exports[`Getting error details should fetch error from 'beforeEach timeout.json' 1`] = `
{
  "codeFrame": {
    "absoluteFile": "/Users/agoldis/currents-playwright/apps/e2e/errorDetails/beforeEach.timeout.spec.ts",
    "column": 8,
    "frame": "  4 | const t = "beforeEach timeout";
  5 | test.describe(t, function () {
> 6 |   test.beforeEach(async () => {
    |        ^
  7 |     await wait(11000);
  8 |   });
  9 |   test(t, async function ({ page }) {",
    "language": "ts",
    "line": 6,
    "originalFile": "/Users/agoldis/currents-playwright/apps/e2e/errorDetails/beforeEach.timeout.spec.ts",
    "relativeFile": null,
  },
  "message": "Test timeout of 1000ms exceeded while running "beforeEach" hook.",
  "name": "Error",
  "stack": "[31mTest timeout of 1000ms exceeded while running "beforeEach" hook.[39m
    at /Users/agoldis/currents-playwright/apps/e2e/errorDetails/beforeEach.timeout.spec.ts:6:8",
}
`;

exports[`Getting error details should fetch error from 'expect timeout.json' 1`] = `
{
  "codeFrame": {
    "absoluteFile": "/Users/agoldis/currents-playwright/apps/e2e/errorDetails/test.timeout.spec.ts",
    "column": 24,
    "frame": "  13 |     test.setTimeout(5 * 60 * 1000);
  14 |     await page.goto("/");
> 15 |     await expect(page).toHaveTitle(/42/);
     |                        ^
  16 |   });
  17 | });
  18 |",
    "language": "ts",
    "line": 15,
    "originalFile": "/Users/agoldis/currents-playwright/apps/e2e/errorDetails/test.timeout.spec.ts",
    "relativeFile": null,
  },
  "message": "Timed out 5000ms waiting for expect(received).toHaveTitle(expected)

Expected pattern: /42/
Received string:  ""
Call log:
  - expect.toHaveTitle with timeout 5000ms
  - waiting for locator(':root')
  -   locator resolved to <html>â€¦</html>
  -   unexpected value ""
  -   locator resolved to <html>â€¦</html>
  -   unexpected value ""
  -   locator resolved to <html>â€¦</html>
  -   unexpected value ""
  -   locator resolved to <html>â€¦</html>
  -   unexpected value ""
  -   locator resolved to <html>â€¦</html>
  -   unexpected value ""
  -   locator resolved to <html>â€¦</html>
  -   unexpected value ""
  -   locator resolved to <html>â€¦</html>
  -   unexpected value ""
  -   locator resolved to <html>â€¦</html>
  -   unexpected value ""
  -   locator resolved to <html>â€¦</html>
  -   unexpected value ""
",
  "name": "Error",
  "stack": "Error: [31mTimed out 5000ms waiting for [39m[2mexpect([22m[31mreceived[39m[2m).[22mtoHaveTitle[2m([22m[32mexpected[39m[2m)[22m

Expected pattern: [32m/42/[39m
Received string:  [31m""[39m
Call log:
  [2m- expect.toHaveTitle with timeout 5000ms[22m
[2m  - waiting for locator(':root')[22m
[2m  -   locator resolved to <html>â€¦</html>[22m
[2m  -   unexpected value ""[22m
[2m  -   locator resolved to <html>â€¦</html>[22m
[2m  -   unexpected value ""[22m
[2m  -   locator resolved to <html>â€¦</html>[22m
[2m  -   unexpected value ""[22m
[2m  -   locator resolved to <html>â€¦</html>[22m
[2m  -   unexpected value ""[22m
[2m  -   locator resolved to <html>â€¦</html>[22m
[2m  -   unexpected value ""[22m
[2m  -   locator resolved to <html>â€¦</html>[22m
[2m  -   unexpected value ""[22m
[2m  -   locator resolved to <html>â€¦</html>[22m
[2m  -   unexpected value ""[22m
[2m  -   locator resolved to <html>â€¦</html>[22m
[2m  -   unexpected value ""[22m
[2m  -   locator resolved to <html>â€¦</html>[22m
[2m  -   unexpected value ""[22m

    at /Users/agoldis/currents-playwright/apps/e2e/errorDetails/test.timeout.spec.ts:15:24",
}
`;

exports[`Getting error details should fetch error from 'global timeout A.json' 1`] = `
{
  "codeFrame": undefined,
  "message": "Test interrupted",
  "name": "Error",
  "stack": "",
}
`;

exports[`Getting error details should fetch error from 'global timeout B.json' 1`] = `null`;

exports[`Getting error details should fetch error from 'global timeout explicitly skipped test.json' 1`] = `null`;
